api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -507,6 +507,11 @@ message HttpConnectionManager {
   // 3. Tracing decision (sampled, forced, etc) is set in 14th byte of the UUID.
   RequestIDExtension request_id_extension = 36;
 
+  // If `preserve_upstream_date` is true, the value of the `date` header sent by the upstream
+  // host will not be overwritten by the HTTP Connection Manager. Otherwise, the HTTP Connection
+  // Manager will overwrite this value unconditionally. The default value is true.
+  google.protobuf.BoolValue preserve_upstream_date = 38;
```

Comment:
Given we believe this is more correct, @mattklein123 WDYT about only having the runtime guard and landing the config knob iff anyone wants to preserve the old behavior?

Alternately if we think it's worth leaving the knob in, we should comment about the runtime guard both here and in docs/root/version_history/current.rst and I think we're good to land.

Reply:
+1 let's just remove the config, have it controlled by runtime, and then we can add config if anyone complains. I find it *highly* unlikely anyone will notice.